### PR TITLE
marconi utxo query: Add query to get the current indexed slot

### PR DESCRIPTION
**Pull request by berewt**
Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [ ] Formatting, PNG optimization, etc. are updated
    - [x] Important changes are reflected in changelog.d of the affected packages
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [ ] Reference the ADR in the PR and reference the PR in the ADR (if revelant)
    - [ ] Reviewer requested

### CODE REVIEW

File: marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/Utxo/UtxoIndex.hs [hunk 1]
```diff
@@ -95,13 +97,29 @@ tests = testGroup "Spec.Marconi.ChainIndex.Indexers.Utxo"
...
           "propJsonRoundtripUtxoRow"
           propJsonRoundtripUtxoRow
+
+    , testGroup "LastSync query"
+        [ testCase
+             "Empty indexer latest sync point is ChainPointAtGenesis"
+             testLastSyncOnFreshIndexer
+
+        , testPropertyNamed
+             "The latest chain point is correct on a indexer that has processed at least a block"
```
Comment (koslambrou) [inline marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/Utxo/UtxoIndex.hs:107]: I have an issue with the term "is correct" :D What is correct? Maybe a description along the lines of: "Querying the latest chain point should return the slot of the last indexed block"?

File: marconi-chain-index/marconi-chain-index.cabal [hunk 2]
```diff
@@ -324,4 +324,5 @@ test-suite marconi-chain-index-test
     , tasty
     , tasty-golden
     , tasty-hedgehog
+    , tasty-hunit
```
[the remark this answers is not in the document]
Comment (berewt) [reply marconi-chain-index/marconi-chain-index.cabal:327]: Yes, but looking at hedgehog doc, it seems that we can lift test and we may not need hunit, I'll try to remove it tomorrow.

File: marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/Utxo/UtxoIndex.hs [hunk 3]
```diff
@@ -95,13 +97,29 @@ tests = testGroup "Spec.Marconi.ChainIndex.Indexers.Utxo"
...
+             "propLastChainPointOnRunningIndexer"
+             propLastChainPointOnRunningIndexer
+
+        , testPropertyNamed
+             "The latest chain point is correct on a indexer that was resumed"
```
Comment (koslambrou) [inline marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/Utxo/UtxoIndex.hs:112]: Maybe a description along the lines of: "When resuming an indexer to a given point, the latest chain point should be that same point"?

File: marconi-chain-index/src/Marconi/ChainIndex/Indexers/Utxo.hs [hunk 4]
```diff
@@ -73,6 +74,18 @@ import Marconi.Core.Storable (Buffered (getStoredEvents, persistToStorage), HasP
                               StorablePoint, StorableQuery, StorableResult, emptyState)
 import Marconi.Core.Storable qualified as Storable
 
+{- Note [Last sync chainpoint]
+ -
+ - The 'LastSyncPoint' query don't return the last indexed chainpoint, but the one before.
```
Comment (koslambrou) [inline marconi-chain-index/src/Marconi/ChainIndex/Indexers/Utxo.hs:79]: don't -> doesn't

**Review by koslambrou — APPROVED** [review]
Can you also update the README of `marconi-sidechain` by removing the "(NOT IMPLEMNETED YET)" line of the query? Also update the description of the method by saying what you wrote in the Note.

**Comment by berewt** [comment]
> Can you also update the README of `marconi-sidechain` by removing the "(NOT IMPLEMNETED YET)" line of the query? Also update the description of the method by saying what you wrote in the Note.

Yes, and in `scriv` (note to self).

**Comment by koslambrou** [comment]
@berewt Merge?

**Pull request merged**